File: packages/python/src/Seguranca/SuiteDeAtaques.py
"""Orquestra a execução de uma coleção de ataques contra um alvo."""

from .ResultadoAtaque import ResultadoAtaque
from .SkipAtaqueException import SkipAtaqueException


class SuiteDeAtaques:
    def __init__(self):
        self._ataques = []

    def adicionar(self, ataque):
        self._ataques.append(ataque)
        return self

    def rodar(self, alvo) -> list:
        resultados = []
        for ataque in self._ataques:
            try:
                resultados.append(ataque.executar(alvo))
            except SkipAtaqueException as e:
                resultados.append(
                    ResultadoAtaque(ataque.nome(), False, "pulado", "Pulado: " + str(e))
                )
            except Exception as e:  # noqa: BLE001 - queremos capturar qualquer falha do ataque
                resultados.append(
                    ResultadoAtaque(ataque.nome(), True, "erro", "Erro ao executar: " + str(e))
                )
        return resultados

    def rodar_e_imprimir(self, alvo) -> bool:
        resultados = self.rodar(alvo)
        vulnerabilidades = 0

        print("=" * 70)
        print("RELATÓRIO DA SUÍTE DE ATAQUES")
        print("=" * 70)
        print()

        for r in resultados:
            print(r.linha_resumo())
            if r.vulneravel and r.severidade not in ("pulado", "demonstracao"):
                vulnerabilidades += 1

        print()
        print("=" * 70)
        if vulnerabilidades == 0:
            print(f"RESUMO: nenhuma vulnerabilidade encontrada em {len(resultados)} ataque(s).")
        else:
            print(
                f"RESUMO: {vulnerabilidades} vulnerabilidade(s) encontrada(s) de "
                f"{len(resultados)} ataque(s) rodados!"
            )
        print("=" * 70)

        return vulnerabilidades == 0

File: packages/php/src/Seguranca/SuiteDeAtaques.php
<?php

namespace Application\Seguranca;

class SuiteDeAtaques
{
    /** @var AtaqueInterface[] */
    private array $ataques = [];

    public function adicionar(AtaqueInterface $ataque): static
    {
        $this->ataques[] = $ataque;
        return $this;
    }

    /** @return ResultadoAtaque[] */
    public function rodar(AlvoCriptografico $alvo): array
    {
        $resultados = [];
        foreach ($this->ataques as $ataque) {
            try {
                $resultados[] = $ataque->executar($alvo);
            } catch (SkipAtaqueException $e) {
                $resultados[] = new ResultadoAtaque(
                    $ataque->nome(),
                    vulneravel: false,
                    severidade: 'pulado',
                    detalhes: 'Pulado: ' . $e->getMessage(),
                );
            } catch (\Throwable $e) {
                $resultados[] = new ResultadoAtaque(
                    $ataque->nome(),
                    vulneravel: true,
                    severidade: 'erro',
                    detalhes: 'Erro ao executar: ' . $e->getMessage(),
                );
            }
        }
        return $resultados;
    }

    /** Roda e imprime um relatório direto no console. */
    public function rodarEImprimir(AlvoCriptografico $alvo): bool
    {
        $resultados = $this->rodar($alvo);
        $vulnerabilidades = 0;

        echo str_repeat('=', 70) . "\n";
        echo "RELATÓRIO DA SUÍTE DE ATAQUES\n";
        echo str_repeat('=', 70) . "\n\n";

        foreach ($resultados as $r) {
            echo $r->linhaResumo() . "\n";
            if ($r->vulneravel && !in_array($r->severidade, ['pulado', 'demonstracao'], true)) {
                $vulnerabilidades++;
            }
        }

        echo "\n" . str_repeat('=', 70) . "\n";
        if ($vulnerabilidades === 0) {
            echo "RESUMO: nenhuma vulnerabilidade encontrada em " . count($resultados) . " ataque(s).\n";
        } else {
            echo "RESUMO: $vulnerabilidades vulnerabilidade(s) encontrada(s) de " . count($resultados) . " ataque(s) rodados!\n";
        }
        echo str_repeat('=', 70) . "\n";

        return $vulnerabilidades === 0;
    }
}
